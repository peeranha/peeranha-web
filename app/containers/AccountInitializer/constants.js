/*
 *
 * AccountInitializer constants
 *
 */

export const GET_CURRENT_ACCOUNT = 'app/AccountInitializer/GET_CURRENT_ACCOUNT';
export const GET_CURRENT_ACCOUNT_SUCCESS =
  'app/AccountInitializer/GET_CURRENT_ACCOUNT_SUCCESS';
export const GET_CURRENT_ACCOUNT_ERROR =
  'app/AccountInitializer/GET_CURRENT_ACCOUNT_ERROR';

export const SELECT_ACCOUNT = 'app/SignUp/SELECT_ACCOUNT';
export const SELECT_ACCOUNT_SUCCESS = 'app/SignUp/SELECT_ACCOUNT_SUCCESS';
export const SELECT_ACCOUNT_ERROR = 'app/SignUp/SELECT_ACCOUNT_ERROR';

export const NO_SCATTER = 'app/SignUp/NO_SCATTER';
export const NO_SELECTED_SCATTER_ACCOUNTS =
  'app/containers/Button/NO_SELECTED_SCATTER_ACCOUNTS';
export const USER_IS_ABSENT_IN_SYSTEM_AND_LOGIN =
  'app/containers/Button/USER_IS_ABSENT_IN_SYSTEM_AND_LOGIN';
export const USER_IS_ABSENT_IN_SYSTEM_AND_SIGNUP =
  'app/containers/Button/USER_IS_ABSENT_IN_SYSTEM_AND_SIGNUP';
