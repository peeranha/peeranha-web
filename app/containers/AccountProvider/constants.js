/*
 *
 * AccountProvider constants
 *
 */

export const GET_CURRENT_ACCOUNT = 'app/AccountProvider/GET_CURRENT_ACCOUNT';
export const GET_CURRENT_ACCOUNT_SUCCESS =
  'app/AccountProvider/GET_CURRENT_ACCOUNT_SUCCESS';
export const GET_CURRENT_ACCOUNT_ERROR =
  'app/AccountProvider/GET_CURRENT_ACCOUNT_ERROR';
