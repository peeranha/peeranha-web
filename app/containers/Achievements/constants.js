export const uniqueAchievementsArr = [
  /**
   * FOUNDING_MEMBER does not depends on rating.
   * This is mock values.
   * This approach allows to cover first rating zone
   *  and
   * to avoid unnecessary elements filtering in achievements selectors
   */

  {
    id: 30,
    title: 'FOUNDING_MEMBER',
    lowerValue: -Infinity,
    upperValue: 99,
    limit: 10000,
  },
  {
    id: 31,
    title: 'ACTIVIST',
    lowerValue: 100,
    upperValue: 499,
    limit: process.env.ENV === 'prod' ? 700 : 5,
  },
  {
    id: 32,
    title: 'RESEARCHER',
    lowerValue: 500,
    upperValue: 999,
    limit: process.env.ENV === 'prod' ? 500 : 5,
  },
  {
    id: 33,
    title: 'HONORABLE_RESIDENT',
    lowerValue: 1000,
    upperValue: 2499,
    limit: process.env.ENV === 'prod' ? 300 : 5,
  },
  {
    id: 34,
    title: 'THE_WISEST',
    lowerValue: 2500,
    upperValue: 4999,
    limit: process.env.ENV === 'prod' ? 150 : 5,
  },
  {
    id: 35,
    title: 'MODERN_HERO',
    lowerValue: 5000,
    upperValue: 9999,
    limit: process.env.ENV === 'prod' ? 50 : 5,
  },
  {
    id: 36,
    title: 'KING_OF_THE_HILL',
    lowerValue: 10000,
    upperValue: Infinity,
    limit: 10,
  },
];

export const achievementsArr = [
  /**
   * WELCOME_STRANGER does not depends on rating.
   * This is mock values.
   * This approach allows to cover first rating zone
   *  and
   * to avoid unnecessary elements filtering in achievements selectors
   */

  {
    id: 40,
    title: 'WELCOME_STRANGER',
    lowerValue: -Infinity,
    upperValue: 99,
  },
  { id: 41, title: 'NEWBIE', lowerValue: 100, upperValue: 499 },
  { id: 42, title: 'JUNIOR', lowerValue: 500, upperValue: 999 },
  { id: 43, title: 'RESIDENT', lowerValue: 1000, upperValue: 2499 },
  { id: 44, title: 'SENIOR', lowerValue: 2500, upperValue: 4999 },
  { id: 45, title: 'HERO', lowerValue: 5000, upperValue: 9999 },
  {
    id: 46,
    title: 'SUPERHERO',
    lowerValue: 10000,
    upperValue: Infinity,
  },
];

export const questionsAskedArr = [
  {
    id: 1,
    title: 'QUESTIONS_ASKED_BRONZE',
    lowerValue: process.env.ENV === 'prod' ? 10 : 2,
    upperValue: process.env.ENV === 'prod' ? 49 : 4,
    level: 'bronze',
  },
  {
    id: 2,
    title: 'QUESTIONS_ASKED_SILVER',
    lowerValue: process.env.ENV === 'prod' ? 50 : 5,
    upperValue: process.env.ENV === 'prod' ? 99 : 9,
    level: 'silver',
  },
  {
    id: 3,
    title: 'QUESTIONS_ASKED_GOLD',
    lowerValue: process.env.ENV === 'prod' ? 100 : 10,
    upperValue: Infinity,
    level: 'gold',
  },
];

export const answerGivenArr = [
  {
    id: 10,
    title: 'ANSWERS_GIVEN_BRONZE',
    lowerValue: process.env.ENV === 'prod' ? 20 : 2,
    upperValue: process.env.ENV === 'prod' ? 99 : 4,
    level: 'bronze',
  },
  {
    id: 11,
    title: 'ANSWERS_GIVEN_SILVER',
    lowerValue: process.env.ENV === 'prod' ? 100 : 5,
    upperValue: process.env.ENV === 'prod' ? 199 : 9,
    level: 'silver',
  },
  {
    id: 12,
    title: 'ANSWERS_GIVEN_GOLD',
    lowerValue: process.env.ENV === 'prod' ? 200 : 10,
    upperValue: Infinity,
    level: 'gold',
  },
];

export const bestAnswerArr = [
  {
    id: 20,
    title: 'BEST_ANSWERS_BRONZE',
    lowerValue: process.env.ENV === 'prod' ? 5 : 1,
    upperValue: process.env.ENV === 'prod' ? 24 : 2,
    level: 'bronze',
  },
  {
    id: 21,
    title: 'BEST_ANSWERS_SILVER',
    lowerValue: process.env.ENV === 'prod' ? 25 : 3,
    upperValue: process.env.ENV === 'prod' ? 49 : 4,
    level: 'silver',
  },
  {
    id: 22,
    title: 'BEST_ANSWERS_GOLD',
    lowerValue: process.env.ENV === 'prod' ? 50 : 5,
    upperValue: Infinity,
    level: 'gold',
  },
];

export const firstIn15Arr = [
  {
    id: 50,
    title: 'ANSWERS_IN_15_BRONZE',
    lowerValue: process.env.ENV === 'prod' ? 5 : 1,
    upperValue: process.env.ENV === 'prod' ? 9 : 2,
    level: 'bronze',
  },
  {
    id: 51,
    title: 'ANSWERS_IN_15_SILVER',
    lowerValue: process.env.ENV === 'prod' ? 10 : 3,
    upperValue: process.env.ENV === 'prod' ? 19 : 4,
    level: 'silver',
  },
  {
    id: 52,
    title: 'ANSWERS_IN_15_GOLD',
    lowerValue: process.env.ENV === 'prod' ? 20 : 5,
    upperValue: Infinity,
    level: 'gold',
  },
];

export const firstAnswerArr = [
  {
    id: 60,
    title: 'FIRST_ANSWERS_BRONZE',
    lowerValue: process.env.ENV === 'prod' ? 10 : 1,
    upperValue: process.env.ENV === 'prod' ? 24 : 2,
    level: 'bronze',
  },
  {
    id: 61,
    title: 'FIRST_ANSWERS_SILVER',
    lowerValue: process.env.ENV === 'prod' ? 25 : 3,
    upperValue: process.env.ENV === 'prod' ? 49 : 4,
    level: 'silver',
  },
  {
    id: 62,
    title: 'FIRST_ANSWERS_GOLD',
    lowerValue: process.env.ENV === 'prod' ? 50 : 5,
    upperValue: Infinity,
    level: 'gold',
  },
];

// achievements groups

export const ratingRelated = 'ratingRelated';
export const uniqueRatingRelated = 'uniqueRatingRelated';
export const questionAskedRelated = 'questionAskedRelated';
export const answerGivenRelated = 'answerGivenRelated';
export const bestAnswerRelated = 'bestAnswerRelated';
export const firstAnswerIn15Related = 'firstAnswerIn15Related';
export const firstAnswerRelated = 'firstAnswerRelated';

// action types

const GET_USER_ACHIEVEMENTS = 'containers/Achievements/GET_USER_ACHIEVEMENTS';
const GET_USER_ACHIEVEMENTS_ERROR =
  'containers/Achievements/GET_USER_ACHIEVEMENTS_ERROR';
const GET_USER_ACHIEVEMENTS_SUCCESS =
  'containers/Achievements/GET_USER_ACHIEVEMENTS_SUCCESS';
const SET_VIEW_PROFILE_ACCOUNT =
  'containers/Achievements/SET_VIEW_PROFILE_ACCOUNT';
const RESET_VIEW_PROFILE_ACCOUNT =
  'containers/Achievements/RESET_VIEW_PROFILE_ACCOUNT';
const USER_ACHIEVEMENTS_LOADING =
  'containers/Achievements/USER_ACHIEVEMENTS_LOADING';
const SET_MEMORIZED_ACHIEV_DATA =
  'containers/Achievements/SET_MEMORIZED_ACHIEV_DATA';
const SET_MAX_GROUPS_LOWER_VALUES =
  'containers/Achievements/SET_MAX_GROUPS_LOWER_VALUES';

export {
  GET_USER_ACHIEVEMENTS,
  GET_USER_ACHIEVEMENTS_ERROR,
  GET_USER_ACHIEVEMENTS_SUCCESS,
  SET_VIEW_PROFILE_ACCOUNT,
  RESET_VIEW_PROFILE_ACCOUNT,
  USER_ACHIEVEMENTS_LOADING,
  SET_MEMORIZED_ACHIEV_DATA,
  SET_MAX_GROUPS_LOWER_VALUES,
};
