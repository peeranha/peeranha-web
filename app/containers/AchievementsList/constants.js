export const possibleAchievements = [
  { id: 1, title: 'QUESTION_ASKED' },
  { id: 2, title: 'ANSWER_GIVEN' },
  { id: 3, title: 'CORRECT_ANSWER' },
  { id: 4, title: 'FIRST_10K_REGISTERED' },
  { id: 5, title: 'STRANGER' },
  { id: 6, title: 'NEWBIE' },
  { id: 7, title: 'JUNIOR' },
  { id: 8, title: 'RESIDENT' },
  { id: 9, title: 'SENIOR' },
  { id: 10, title: 'HERO' },
  { id: 11, title: 'SUPERHERO' },
];

const GET_USER_ACHIEVEMENTS =
  'containers/AchievementsList/GET_USER_ACHIEVEMENTS';
const GET_USER_ACHIEVEMENTS_ERROR =
  'containers/AchievementsList/GET_USER_ACHIEVEMENTS_ERROR';
const GET_USER_ACHIEVEMENTS_SUCCESS =
  'containers/AchievementsList/GET_USER_ACHIEVEMENTS_SUCCESS';

export {
  GET_USER_ACHIEVEMENTS,
  GET_USER_ACHIEVEMENTS_ERROR,
  GET_USER_ACHIEVEMENTS_SUCCESS,
};
