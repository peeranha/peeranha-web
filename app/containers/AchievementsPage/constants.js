export const possibleAchievements = [
  { id: 1, title: 'QUESTION_ASKED' },
  { id: 2, title: 'ANSWER_GIVEN' },
  { id: 3, title: 'CORRECT_ANSWER' },
  { id: 4, title: 'FIRST_10K_REGISTERED' },
  { id: 5, title: 'STRANGER' },
  { id: 6, title: 'NEWBIE' },
  { id: 7, title: 'JUNIOR' },
  { id: 8, title: 'RESIDENT' },
  { id: 9, title: 'SENIOR' },
  { id: 10, title: 'HERO' },
  { id: 11, title: 'SUPERHERO' },
];

const GET_USER_ACHIEVEMENTS =
  'containers/AchievementsPage/GET_USER_ACHIEVEMENTS';
const GET_USER_ACHIEVEMENTS_ERROR =
  'containers/AchievementsPage/GET_USER_ACHIEVEMENTS_ERROR';
const GET_USER_ACHIEVEMENTS_SUCCESS =
  'containers/AchievementsPage/GET_USER_ACHIEVEMENTS_SUCCESS';
const SET_CURRENT_ACCOUNT = 'containers/AchievementsPage/SET_CURRENT_ACCOUNT';

export {
  GET_USER_ACHIEVEMENTS,
  GET_USER_ACHIEVEMENTS_ERROR,
  GET_USER_ACHIEVEMENTS_SUCCESS,
  SET_CURRENT_ACCOUNT,
};
