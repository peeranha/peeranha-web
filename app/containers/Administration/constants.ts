const PATH = 'app/Administration/';

export const ADD_MODERATOR_BUTTON_BUTTON =
  'app_Administration_ADD_MODERATOR_BUTTON';
export const OPEN_ADD_MODERATOR_FORM_BUTTON =
  'app_Administration_OPEN_ADD_MODERATOR_FORM_BUTTON';

export const WALLET_ADDRESS_FIELD = `${PATH}WALLET_ADDRESS_FIELD`;

export const GET_MODERATORS = `${PATH}GET_MODERATORS`;
export const GET_MODERATORS_SUCCESS = `${PATH}GET_MODERATORS_SUCCESS`;
export const GET_MODERATORS_ERROR = `${PATH}GET_MODERATORS_ERROR`;

export const ADD_MODERATOR = `${PATH}ADD_MODERATOR`;
export const ADD_MODERATOR_SUCCESS = `${PATH}ADD_MODERATOR_SUCCESS`;
export const ADD_MODERATOR_ERROR = `${PATH}ADD_MODERATOR_ERROR`;

export const REVOKE_MODERATOR = `${PATH}REVOKE_MODERATOR`;
export const REVOKE_MODERATOR_SUCCESS = `${PATH}REVOKE_MODERATOR_SUCCESS`;
export const REVOKE_MODERATOR_ERROR = `${PATH}REVOKE_MODERATOR_ERROR`;
