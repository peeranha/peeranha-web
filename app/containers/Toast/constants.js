/*
 *
 * Toast constants
 *
 */

export const ADD_TOAST = 'app/Toast/ADD_TOAST';
export const REMOVE_TOAST = 'app/Toast/REMOVE_TOAST';

export const REMOVE_TIMEOUT = 5000;

export const BOTTOM_LEFT = 'app/Toast/BOTTOM_LEFT';
export const BOTTOM_RIGHT = 'app/Toast/BOTTOM_RIGHT';
export const TOP_RIGHT = 'app/Toast/TOP_RIGHT';
export const TOP_LEFT = 'app/Toast/TOP_LEFT';
