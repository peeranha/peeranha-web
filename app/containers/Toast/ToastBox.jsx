import styled from 'styled-components';

import { BOTTOM_RIGHT, BOTTOM_LEFT, TOP_RIGHT, TOP_LEFT } from './constants';

const toastsLocation = {
  [TOP_LEFT]: `
    left: 0;
    flex-direction: column;
  `,
  [BOTTOM_LEFT]: `
    left: 0;
    flex-direction: column-reverse;
  `,
  [TOP_RIGHT]: `
    right: 0;
    flex-direction: column;
  `,
  [BOTTOM_RIGHT]: `
    right: 0;
    flex-direction: column-reverse;
  `,
};

const ToastBox = styled.div`
  z-index: 9999999999;
  position: fixed;
  display: flex;
  max-height: 100vh;
  ${x => toastsLocation[x.location]};
`;

export default ToastBox;
