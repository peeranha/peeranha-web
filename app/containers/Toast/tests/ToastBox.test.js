import React from 'react';
import { shallow } from 'enzyme';

import ToastBox from '../ToastBox';
import { TOP_LEFT } from '../constants';

describe('<ToastBox />', () => {
  it('snapshot test', () => {
    const props = { location: TOP_LEFT };
    const renderedComponent = shallow(<ToastBox {...props} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
