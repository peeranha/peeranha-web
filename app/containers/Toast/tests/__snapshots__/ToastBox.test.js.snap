// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToastBox /> snapshot test 1`] = `
<div
  className="ToastBox-IqGQB iHOLbj"
/>
`;
